src/test/java/uk/gov/hmcts/reform/bulkscan/orchestrator/services/ccd/CcdNewCaseCreatorTest.java

```diff
@@ -103,32 +111,59 @@ void should_return_new_case_id_when_successfully_executed_all_the_steps() {
 
         // when
         CreateCaseResult result =
-            ccdNewCaseCreator
-                .createNewCase(
-                    exceptionRecord,
-                    configItem,
-                    true,
-                    IDAM_TOKEN,
-                    USER_ID
-                );
+            ccdNewCaseCreator.createNewCase(exceptionRecord, configItem, true, IDAM_TOKEN, USER_ID);
 
         // then
         assertThat(result.caseId).isEqualTo(CASE_ID);
 
         // and
-        var caseDetailsBuilderCaptor = ArgumentCaptor.forClass(Function.class);
+        var expectedLogContext = format(
+            "Exception ID: %s, jurisdiction: %s, form type: %s",
+            exceptionRecord.id,
+            exceptionRecord.poBoxJurisdiction,
+            exceptionRecord.formType
+        );
+
         verify(ccdApi).createCase(
             eq(IDAM_TOKEN),
-            anyString(),
+            eq(s2sToken),
             eq(USER_ID),
             eq(exceptionRecord.poBoxJurisdiction),
-            anyString(),
-            anyString(),
-            caseDetailsBuilderCaptor.capture(),
-            anyString()
+            eq(expectedTransformationResponse.caseCreationDetails.caseTypeId),
+            eq(expectedTransformationResponse.caseCreationDetails.eventId),
+            any(),
+            eq(expectedLogContext)
+        );
+    }
+
+    @SuppressWarnings("unchecked")
+    @ParameterizedTest
+    @ValueSource(booleans = {true, false})
+    void should_provide_correct_case_data_for_ccd(boolean isServiceEnabledForAutoCaseCreation) {
+        // given
+        var transformationResponse = new SuccessfulTransformationResponse(
+            new CaseCreationDetails("case_type_1", "event_id_1", basicCaseData()),
+            emptyList()
+        );
+
+        given(s2sTokenGenerator.generate()).willReturn("s2s-token1");
+        given(exceptionRecordTransformer.transformExceptionRecord(any(), any())).willReturn(transformationResponse);
+
+        ExceptionRecord exceptionRecord = getExceptionRecord();
+
+        // trigger case creation to capture data content builder which was passed to CCD client
+        ccdNewCaseCreator.createNewCase(exceptionRecord, getConfigItem(), true, IDAM_TOKEN, USER_ID);
+
+        var caseDetailsBuilderCaptor = ArgumentCaptor.forClass(Function.class);
+        verify(ccdApi).createCase(any(), any(), any(), any(), any(), any(), caseDetailsBuilderCaptor.capture(), any());
+
+        // then
+        assertCaseDataContentBuilderCreatesCorrectResult(
+            caseDetailsBuilderCaptor.getValue(),
+            exceptionRecord.id,
+            transformationResponse,
+            isServiceEnabledForAutoCaseCreation
         );
```

Comment:
This kind of test is necessary, because the tested class passes a function to CCD client, which is then executed by it. The correctness of this function must be verified.